### PR TITLE
feat: blocked squad members list

### PR DESCRIPTION
**Pull request by sshanzel**
### Describe what this PR does
- Blocked members list
- Update the UserListModal to accept a header than just a title.
- Add the new parameter for fetching members by role
- Updated the modal tabs component to accept on click callback.
- Introduce usage of `roleRank`.

Preview:

image

image

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-1188 #done

### CODE REVIEW

File: packages/shared/src/components/squads/SquadMemberActions.tsx
Comment (sshanzel) [inline packages/shared/src/components/squads/SquadMemberActions.tsx:56]: Subject to change once the indication PR is merged.

File: packages/shared/src/components/modals/SquadMemberModal.tsx
Comment (rebelchris) [inline packages/shared/src/components/modals/SquadMemberModal.tsx:63]: One side of me likes this, other side feels like it's a bit hacky around the modal guideline.
Not sure if we should change or not, maybe @capJavert has an opinion here?

File: packages/shared/src/components/squads/SquadMemberActions.tsx
Comment (rebelchris) [inline packages/shared/src/components/squads/SquadMemberActions.tsx:27]: Has a tooltip in sketch

**Review by rebelchris — CHANGES_REQUESTED** [review]
Just noting it's blocked until the other one is merged I guess so we can complete that code section.

File: packages/shared/src/components/modals/SquadMemberModal.tsx
Comment (sshanzel) [reply packages/shared/src/components/modals/SquadMemberModal.tsx:63]: I agree with your sentiment. One part makes it quick and working, on the other side, we are not really "switching" the UI per tab as the guideline would suggest. I am willing to refactor if we decide to do two separate components for the list.

File: packages/shared/src/components/modals/SquadMemberModal.tsx
Comment (sshanzel) [reply packages/shared/src/components/modals/SquadMemberModal.tsx:79]: Yes, we do, we keep the `title` property on line 77 🥳

File: packages/shared/src/components/squads/SquadMemberActions.tsx
Comment (sshanzel) [reply packages/shared/src/components/squads/SquadMemberActions.tsx:27]: Great catch! Thank you!

File: packages/shared/src/components/squads/SquadMemberActions.tsx
Comment (rebelchris) [inline packages/shared/src/components/squads/SquadMemberActions.tsx:31]: Minor thing, but think now we mainly do something like this:
```
if (await showPrompt(promptOptions)) {
```

**Review by rebelchris — CHANGES_REQUESTED** [review]
Amazing work, have some minor suggestions

File: packages/shared/src/components/modals/SquadMemberModal.tsx
Comment (sshanzel) [reply packages/shared/src/components/modals/SquadMemberModal.tsx:109]: It is centered but it looks a bit off and a bit better with the additional spacing. WDYT?

Without bottom padding:

image

File: packages/shared/src/components/squads/SquadMemberActions.tsx
Comment (sshanzel) [reply packages/shared/src/components/squads/SquadMemberActions.tsx:33]: This should be correct. We are referring to the selected user in the member's blocked list not the logged in user.

File: packages/shared/src/components/squads/SquadMemberRoleBadge.tsx
Comment (sshanzel) [reply packages/shared/src/components/squads/SquadMemberRoleBadge.tsx:47]: I was also shocked this got past through the lint. Probably happens when using at `classNames` function?

File: packages/shared/src/components/squads/SquadMemberRoleBadge.tsx
Comment (rebelchris) [reply packages/shared/src/components/squads/SquadMemberRoleBadge.tsx:30]: Hmm is there any way we can uniform them into 1 component that can handle both cases?
EG > one can detect role early?

File: packages/shared/src/components/squads/SquadMemberActions.tsx
Comment (sshanzel) [reply packages/shared/src/components/squads/SquadMemberActions.tsx:31]: Ah, just my personal preference, I kinda don't like seeing `await` inside `if` conditions.

Though I think we do this when we are reusing the options, but to keep consistency let me update this one.

File: packages/shared/src/components/modals/SquadMemberModal.tsx
Comment (rebelchris) [reply packages/shared/src/components/modals/SquadMemberModal.tsx:109]: Rather consult with Tomer before changing it :)

File: packages/shared/src/components/squads/SquadMemberActions.tsx
Comment (rebelchris) [reply packages/shared/src/components/squads/SquadMemberActions.tsx:33]: But you deconstruct this right above no?

File: packages/shared/src/components/squads/SquadMemberRoleBadge.tsx
Comment (sshanzel) [reply packages/shared/src/components/squads/SquadMemberRoleBadge.tsx:30]: They have different prop requirements. The other one requires a lot of objects to determine the role, the new one requires just the role which is used by the old one.

File: packages/shared/src/components/squads/SquadMemberActions.tsx
Comment (sshanzel) [reply packages/shared/src/components/squads/SquadMemberActions.tsx:33]: Yes, though that deconstructed value came from context, and the member came from the prop.

Comment (sshanzel) [reply packages/shared/src/components/squads/SquadMemberActions.tsx:33]: Oh, wait. I just checked the code, yep, it seems to be accurate using `user`. Good catch!

File: packages/shared/src/components/squads/SquadMemberActions.tsx
Comment (rebelchris) [reply packages/shared/src/components/squads/SquadMemberActions.tsx:31]: Yeah either way has it's charm, maybe something we can create a rule for?

File: packages/shared/src/components/squads/SquadMemberRoleBadge.tsx
Comment (rebelchris) [reply packages/shared/src/components/squads/SquadMemberRoleBadge.tsx:30]: Hmm, but you could abstract the role in component above and render `SquadMemberBadge` with the role directly?
It feels a bit weird having 2 exports now and the naming is a bit confusing.

Thinking about looking at this 3 months down the line people might use the wrong component

Comment (sshanzel) [reply packages/shared/src/components/squads/SquadMemberRoleBadge.tsx:30]: The reason for the 2 exports is to support both use cases. One for simply showing the appropriate badge for anyone that can provide a role. The other one is for displaying it in the comment section where it uses a hook that abstracted something and has more requirements.

Probably we should rename the old component to be more accurate that it was meant for comment section.

Comment (capJavert) [reply packages/shared/src/components/squads/SquadMemberRoleBadge.tsx:30]: I like the wrapper solution but I think it is better to split those two into two separate files, I don't think we use these kind of mixed exports in other parts of the project?

File: packages/shared/src/graphql/fragments.ts
Comment (rebelchris) [inline packages/shared/src/graphql/fragments.ts:71]: Will have some issue with current fragment changes

File: packages/shared/src/graphql/sources.ts
Comment (rebelchris) [inline packages/shared/src/graphql/sources.ts:28]: I think @capJavert removed this number exposure now.
Maybe should align that?

File: packages/shared/src/components/squads/SquadMemberItemAdditionalContent.tsx
Comment (rebelchris) [inline packages/shared/src/components/squads/SquadMemberItemAdditionalContent.tsx:31]: This can be outside the function (optional comment)

File: packages/shared/src/components/squads/SquadMemberAuthorBadge.tsx
Comment (rebelchris) [inline packages/shared/src/components/squads/SquadMemberAuthorBadge.tsx:55]: Think we should abstract to 2 component, and not 100% sure on naming here.
but might be because I'm missing context of this change now.

**Review by rebelchris — CHANGES_REQUESTED** [review]

File: packages/shared/src/graphql/sources.ts
Comment (sshanzel) [reply packages/shared/src/graphql/sources.ts:28]: Oh, why did we remove it?

File: packages/shared/src/components/squads/SquadMemberItemAdditionalContent.tsx
Comment (sshanzel) [reply packages/shared/src/components/squads/SquadMemberItemAdditionalContent.tsx:31]: I think we can keep this here, so JS won't have to create this object until necessary.

File: packages/shared/src/graphql/fragments.ts
Comment (sshanzel) [reply packages/shared/src/graphql/fragments.ts:71]: I'll check if there is any conflict and coordinate with Ante if necessary.

File: packages/shared/src/components/squads/SquadMemberAuthorBadge.tsx
Comment (sshanzel) [reply packages/shared/src/components/squads/SquadMemberAuthorBadge.tsx:55]: What would we want to be abstracted into 2 components? I mean the content of this component is just 2 lines of code. Do you want me to delete this component and just call the hook on where this was rendered?

File: packages/shared/src/graphql/sources.ts
Comment (capJavert) [reply packages/shared/src/graphql/sources.ts:28]: I removed the one for posting permission (different field on Source). But generally I think frontend does not use numeric ranks anywhere, all checks are done through permissions so maybe this is not needed too.

File: packages/shared/src/components/modals/SquadMemberModal.tsx
Comment (capJavert) [reply packages/shared/src/components/modals/SquadMemberModal.tsx:81]: Should we not check `verifyPermission(squad, SourcePermissions.ViewBlockedMembers)` instead?

Comment (capJavert) [reply packages/shared/src/components/modals/SquadMemberModal.tsx:81]: This would also remove the need for frontend to know about role ranks.

File: packages/shared/src/components/modals/SquadMemberModal.tsx
Comment (capJavert) [reply packages/shared/src/components/modals/SquadMemberModal.tsx:64]: Sorry, missed this, what guideline are we talking about specifically?

File: packages/shared/src/components/squads/SquadMemberAuthorBadge.tsx
Comment (rebelchris) [reply packages/shared/src/components/squads/SquadMemberAuthorBadge.tsx:55]: Let's go with the second option in that case.
Just to avoid confusion couple months down the line.

File: packages/shared/src/components/modals/SquadMemberModal.tsx
Comment (rebelchris) [reply packages/shared/src/components/modals/SquadMemberModal.tsx:81]: Good point, sounds like a good plan to me, wdyt @sshanzel ?

File: packages/shared/src/graphql/sources.ts
Comment (sshanzel) [reply packages/shared/src/graphql/sources.ts:28]: Oh, cool. Now that we have a use case here, we can now probably keep it.

Comment (rebelchris) [reply packages/shared/src/graphql/sources.ts:28]: @sshanzel but is it no better to evaluate actual permissions like Ante suggested to check if someone can view blocked members? (Seems more inline with what we do)

Comment (sshanzel) [reply packages/shared/src/graphql/sources.ts:28]: Oh, right, we have a new permission for blocked members. We can probably use that.

File: packages/shared/src/components/modals/SquadMemberModal.tsx
Comment (rebelchris) [reply packages/shared/src/components/modals/SquadMemberModal.tsx:70]: Mainly modal guideline, normally the stepped modal expect 2 seperate views.
Here we technically "hacked" a workaround for updates outside the modal.

**Review by rebelchris — APPROVED** [review]
Yeah, thinks this is more explicit and less error prone.
If we see more use for this role badge we can abstract into custom component.

File: packages/shared/src/components/modals/SquadMemberModal.tsx
Comment (sshanzel) [reply packages/shared/src/components/modals/SquadMemberModal.tsx]: Oh, just saw this now. Yes this should be the most updated implementation. Thank you both!

File: packages/shared/src/components/modals/SquadMemberModal.tsx
Comment (capJavert) [reply packages/shared/src/components/modals/SquadMemberModal.tsx:70]: If in all other places we do it with two views then this one should be no exception. But don't have the hard preference because above is simple logic wise.

**Review by capJavert — APPROVED** [review]

**Pull request merged**
